src/main/java/org/springframework/data/jpa/repository/query/ParameterMetadataProvider.java

```diff
@@ -185,19 +188,21 @@ private ParameterMetadataProvider(CriteriaBuilder builder, @Nullable Iterator<Ob
 
 		static final Object PLACEHOLDER = new Object();
 
+		private final Part part;
 		private final Type type;
 		private final ParameterExpression<T> expression;
 		private final PersistenceProvider persistenceProvider;
 
 		/**
 		 * Creates a new {@link ParameterMetadata}.
 		 */
-		public ParameterMetadata(ParameterExpression<T> expression, Type type, @Nullable Object value,
```

Comment:
Can you clarify why this change is needed? I don't feel comfortable with making `ParameterMetadata` aware of anything ignore-case related. Shouldn't the handling of that be applied in the code eventually calling `….prepare(…)`?

Reply:
I need more time to study this proposal

Reply:
I can't find better solution. Method .prepare calling only from org.springframework.data.jpa.repository.query.QueryParameterSetterFactory.CriteriaQueryParameterSetterFactory#CriteriaQueryParameterSetterFactory, but there have not place for this. In ParameterMetadata we prepare another string conversion and can do this conversion. I think that I should change ignoreCaseType field to boolean